src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -325,60 +325,57 @@ CancellationToken cancellationToken
 		}
 
 		private static void AnalyzeConflictingImmutabilityOnTypeParameters(
-			SyntaxNodeAnalysisContext ctx,
+			SymbolAnalysisContext ctx,
+			INamedTypeSymbol namedTypeSymbol,
 			AnnotationsContext annotationsContext
 		) {
-			// Get the symbol for the parameter
-			if( ctx.SemanticModel.GetDeclaredSymbol( ctx.Node, ctx.CancellationToken ) is not ITypeParameterSymbol symbol ) {
-				return;
-			}
+			foreach( var parameter in namedTypeSymbol.TypeParameters ) {
+				// Check if the parameter has both the [Immutable] and the [OnlyIf] attributes
+				if( !annotationsContext.Objects.Immutable.IsDefined( parameter )
+					|| !annotationsContext.Objects.OnlyIf.IsDefined( parameter )
+				) {
+					return;
+				}
 
-			// Check if the parameter has both the [Immutable] and the [OnlyIf] attributes
-			if( !annotationsContext.Objects.Immutable.IsDefined( symbol )
-				|| !annotationsContext.Objects.OnlyIf.IsDefined( symbol )
-			) {
-				return;
+				// Create the diagnostic on the parameter (excluding the attribute)
+				ctx.ReportDiagnostic(
+					Diagnostics.ConflictingImmutability,
+					parameter.Locations[0],
+					messageArgs: new[] {
+						"Immutable",
+						"ConditionallyImmutable.OnlyIf",
+						"typeparameter"
+					}
+				);
 			}
-
-			// Create the diagnostic on the parameter (excluding the attribute)
-			ctx.ReportDiagnostic(
-				Diagnostics.ConflictingImmutability,
-				symbol.DeclaringSyntaxReferences[ 0 ].GetSyntax( ctx.CancellationToken ).GetLastToken().GetLocation(),
-				messageArgs: new[] {
-					"Immutable",
-					"ConditionallyImmutable.OnlyIf",
-					symbol.Kind.ToString().ToLower()
-				}
-			);
 		}
 
 		private static void AnalyzeConflictingImmutabilityOnMember(
-			SyntaxNodeAnalysisContext ctx,
+			SymbolAnalysisContext ctx,
+			INamedTypeSymbol symbol,
 			AnnotationsContext annotationsContext
 		) {
-
-			// Ensure syntax is expected and get the symbol
-			if( ctx.Node is not TypeDeclarationSyntax syntax ) {
-				return;
-			}
-			var symbol = ctx.SemanticModel.GetDeclaredSymbol( syntax, ctx.CancellationToken );
-
 			// Get information about immutability
 			bool hasImmutable = annotationsContext.Objects.Immutable.IsDefined( symbol );
 			bool hasConditionallyImmutable = annotationsContext.Objects.ConditionallyImmutable.IsDefined( symbol );
 			bool hasImmutableBase = annotationsContext.Objects.ImmutableBaseClass.IsDefined( symbol );
 
+			TypeDeclarationSyntax syntax = null;
+			TypeDeclarationSyntax GetSyntax() {
+				return syntax ??= (TypeDeclarationSyntax)symbol.DeclaringSyntaxReferences[ 0 ].GetSyntax( ctx.CancellationToken );
+			}
+
 			// Check if there are conflicting immutability attributes
 			if( hasImmutable && hasConditionallyImmutable ) {
 				// [Immutable] and [ConditionallyImmutable] both exist,
 				// so create a diagnostic
 				ctx.ReportDiagnostic(
 					Diagnostics.ConflictingImmutability,
-					syntax.Identifier.GetLocation(),
+					GetSyntax().Identifier.GetLocation(),

```

Comment:
I checked afterward and `symbol.Locations[ 0 ]` is actually the identifier location, so grabbing the syntax isn't actually necessary.

`Locations[0]` and `DeclaringSyntaxReferences[0]` of course both have the problem of "it might not be the _right_ partial" with the attributes on it.

Reply:
Err actually, existing setup has that problem anyway and would create multiple diagnostics for partials.